### PR TITLE
GH-39257: [JS] LargeBinary

### LINKED ISSUE #39257: [JS] Support LargeBinary type
### Describe the enhancement requested

Support the data type. 

### Component(s)

JavaScript

### PR DESCRIPTION
**Pull request by domoritz**
Merge after #39249
* Closes: #39257

### CODE REVIEW

File: js/src/builder/largebinary.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                offsets.set(index, BigInt(0));
+            } else {
+                const length = value.length;
+                data.set(value, offset);
+                offsets.set(index, BigInt(length));
+                offset += length;
+            }
+        }
+    }
+}
```
Comment (bmschmidt) [inline js/src/builder/largebinary.ts:54]: Do you want to activate the placeholder in builder/largeutf8.ts in this PR? 

Seems like you could probably do the same for getByteLength, too.

```
(LargeUtf8Builder.prototype as any)._flushPending = (LargeBinaryBuilder.prototype as any)._flushPending;
```

**Review by bmschmidt — APPROVED** [review]
Looks good to me, fwiw, except for the duplicated code between the utf8 and binary builders. The  new unification stuff is nice.

**Comment by domoritz** [comment]
Oh good catch. I forgot to remove that duplicate code.

**Comment by domoritz** [comment]
CC @kylebarron

**Review by kylebarron — APPROVED** [review]
lgtm!

File: js/src/builder/largebinary.ts (hunk 1 above)
Comment (domoritz) [reply js/src/builder/largebinary.ts:54]: Yes, just updated the pull request.

**Pull request merged**